src/Stratis.Features.FederatedPeg/FederatedPegFeature.cs

```diff
@@ -333,7 +333,7 @@ public static IFullNodeBuilder UseFederatedPegPoAMining(this IFullNodeBuilder fu
                 features.AddFeature<PoAFeature>().DependOn<FederatedPegFeature>().FeatureServices(services =>
                     {
                         services.AddSingleton<PoABlockHeaderValidator>();
-                        services.AddSingleton<IPoAMiner, PoAMiner>();
+                        services.AddSingleton<IPoAMiner, CollateralPoAMiner>();
```

Comment:
Does `CollateralChecker` need to be injected too?

Reply:
Collateral checker is already on DI